compliance/collection/main.go

```diff
@@ -37,6 +37,18 @@ var (
 	once sync.Once
 )
 
+func nodeInventoryWaitCallback(waitTime time.Duration) {
+	maxBackoffSeconds := int64(env.NodeInventoryMaxBackoff.DurationSetting() / time.Second)
+	waitTimeSeconds := int64(waitTime / time.Second)
+
+	if waitTimeSeconds > maxBackoffSeconds {
+		log.Debugf("Backoff interval hit upper boundary. Cutting from %d to %d", waitTimeSeconds, maxBackoffSeconds)
+		waitTimeSeconds = maxBackoffSeconds
+	}
```

Comment:
This fragment seems redundant. Maybe other things in this `nodeInventoryWaitCallback` need cleanup too?